packages/core/src/browser/widget-manager.ts

```diff
@@ -73,18 +74,21 @@ export class WidgetManager {
      */
     async getOrCreateWidget<T extends Widget>(factoryId: string, options?: any): Promise<T> {
         const key = this.toKey({ factoryId, options });
-        const existingWidget = this.widgets.get(key);
+        const existingWidget = this.widgetPromises.get(key);
```

Comment:
@hexa00 here we check whether a request exists

Reply:
Haaa someone I missed that piece of code , thx! :)